sovrin_client/agent/agent.py

```diff
@@ -202,12 +217,73 @@ def _initIssuerProverVerifier(self):
         self.prover = SovrinProver(client=self.client, wallet=self._wallet)
         self.verifier = SovrinVerifier(client=self.client, wallet=self._wallet)
 
+    @property
+    def wallet(self):
+        return self._wallet
+
+    @wallet.setter
+    def wallet(self, newWallet):
+        self._wallet = newWallet
+
     @Agent.client.setter
     def client(self, client):
         Agent.client.fset(self, client)
         if self.client:
             self._initIssuerProverVerifier()
 
+    def getContextDir(self):
+        # TODO: Will we need any environment context for agent's walletS?
+        return os.path.expanduser(os.path.join(
+            self.config.baseDir, self.config.keyringsDir, "agents",
+            self.name.lower().replace(" ", "-")))
+
+    def start(self, loop):
+        super().start(loop)
+        self._restoreLastActiveWallet()
```

Comment:
If we have created an instance of this class and say user passed a wallet. In constructor, user passed wallet will be set as active wallet. However, as soon as start method is called. It will override user passed wallet with previously saved wallet.

I think we should not restore it here.